### DISCUSSION TITLE
Some beatmaps don't seem to work properly in the editor

### DISCUSSION THREAD

**Discussion by Jan-Arbus** [ppy/osu#20238] [Q&A] [answered]
Timelines stacked together
If you select an object and then deselect it, most of the UI becomes a black block
This is the beatmap in question https://osu.ppy.sh/beatmapsets/39804#osu/126645
![屏幕截图 2022-09-11 190136](https://user-images.githubusercontent.com/74862110/189523995-56920b1e-508e-49e7-b4e0-d887244b9f21.png)
[updater.log](https://github.com/ppy/osu/files/9542328/updater.log)
[runtime.log](https://github.com/ppy/osu/files/9542329/runtime.log)
[network.log](https://github.com/ppy/osu/files/9542330/network.log)
[performance.log](https://github.com/ppy/osu/files/9542331/performance.log)

**Comment by smoogipoo** [comment]
I can't repro this, but when it happens can you go to the bottom of settings and try enabling this setting?
![image](https://user-images.githubusercontent.com/1329837/189570471-6203ebec-6148-4eed-9874-1e0c774c64cf.png)

**Reply by Jan-Arbus** [reply]
> I can't repro this, but when it happens can you go to the bottom of settings and try enabling this setting? ![image](https://user-images.githubusercontent.com/1329837/189570471-6203ebec-6148-4eed-9874-1e0c774c64cf.png)

After enabling this option, those blacked out UI disappeared :(

**Reply by smoogipoo** [reply]
Is that to say that the game looks normal?

**Reply by Jan-Arbus** [reply]
not
![屏幕截图 2022-09-12 124006](https://user-images.githubusercontent.com/74862110/189575192-1ed26195-867d-42a3-bf2d-c70cead805fb.png)

**Comment by smoogipoo** [comment]
Are your GPU drivers up to date? Please provide your [dxdiag output](https://support.microsoft.com/en-us/help/4028644/windows-open-and-run-dxdiagexe).

**Reply by Jan-Arbus** [reply]
The GPU driver is not up to date, I will try to upgrade the driver later
[DxDiag.txt](https://github.com/ppy/osu/files/9544699/DxDiag.txt)

**Reply by Jan-Arbus** [reply]
Unfortunately, updating the driver to the latest version didn't help, the problem persists

**Reply by smoogipoo** [reply]
Not sure. I think this is probably going to be better when we change to using Veldrid/DirectX for your system.

**Comment by Jan-Arbus** [comment]
Well, I found out that the problem was with my local beatmaps
He was migrated from a very old version
[beatmap.zip](https://github.com/ppy/osu/files/9563347/beatmap.zip)

**Reply by smoogipoo** [reply]
Thanks for this. Initially I was going to say beatmaps can't cause what you're seeing, but part of the issue is reproducible via the `Another` diff, as seen here:

![image](https://user-images.githubusercontent.com/1329837/190086570-b29776df-938e-4a12-b8d7-26b85c9041a3.png)

The rest of my UI doesn't disappear like yours, but that is likely related to the breakage from the front-to-back render pass as mentioned above. Also all hitobjects are seemingly stacked at the same time value on the timeline.

**Comment by smoogipoo** [comment, marked as answer]
Split this into two issues:
- Everything being stacked at one position in the timeline: https://github.com/ppy/osu/issues/20306
- UI disappearing/black areas: https://github.com/ppy/osu-framework/issues/5408
